pkg/model/sdk_helper.go

```diff
@@ -264,7 +267,13 @@ func getMemberByPath(
 		if x == last {
 			return shapeRef, true
 		}
-		shape = shapeRef.Shape
+		elemType := shapeRef.Shape.Type
+		switch elemType {
+		case "list":
+			shape = shapeRef.Shape.MemberRef.Shape
```

Comment:
Eventually we should support `map` type as well, which would be:

```go
case "map":
    shape = shapeRef.Shape.ValueRef.Shape
```

Reply:
You can do that in a followup PR, though, since this one is for lists only (and dealing with map types in the code generator is a little tougher anyway)

Reply:
Sounds good, let me create an issue for that such that it is tracked.